pkg/phlare/phlare.go

```diff
@@ -64,6 +77,8 @@ type Config struct {
 	ConfigFile      string `yaml:"-"`
 	ShowVersion     bool   `yaml:"-"`
 	ConfigExpandEnv bool   `yaml:"-"`
+
+	MainFlags mainFlags `yaml:"-"`
```

Comment:
I would prefer, if we would solve it a bit more like mimir's main here: https://github.com/grafana/mimir/blob/7b143049e16847e6fe31c97cb49d521a8165d25a/cmd/mimir/main.go#LL77

I think removing `ShowVersion` and `MainFlags` and handling it all locally in `main()` sounds a lot cleaner to me. What do you think?

Reply:
I tried it before. I think it is a little bit difficult to follow the mimir approach. Because we have a `DynamicUnmarshal` fn to unmarshal the flags and configs. And we reregister the flags here. https://github.com/grafana/phlare/blob/524465ec82dffe2c2bbc0c16132e262c900af141/pkg/cfg/files.go#L75

Then the main args were reported as unknown args. https://github.com/grafana/phlare/blob/524465ec82dffe2c2bbc0c16132e262c900af141/pkg/cfg/files.go#L87 Then the whole process exit.

I think a way to solve this problem is we also pass the main args to `DynamicUnmarshal` and reregister it too, but it looks ugly to me.

Do you have any ideas about how to fix this issue? I also tried to parse the main args first. But still can not just pass the valid args to `DynamicUnmarshal`, because there is no easy way to ignore main args from os.Args.

Reply:
Also, I think this would not help. https://github.com/grafana/phlare/pull/497#discussion_r1112102226

Because in `DynamicUnmarshal ` we just use a new flag. So we still missing the main flags on this new flag command.
https://github.com/grafana/phlare/blob/524465ec82dffe2c2bbc0c16132e262c900af141/pkg/cfg/files.go#L75
This fresh flag has not had the main flags. But `args []string` has values of main flags.

Reply:
I have clearly not seen the full complexity by just reviewing :slightly_smiling_face:.  

I have given it a try myself, I think it is slightly cleaner than the state of your PR. What do you think?

https://github.com/simonswine/phlare/commit/69bcb553

Reply:
Looks good! I'll follow your approach to update it. Thanks a lot!